core/AssetManager/UIAsset/OnDiskUIAsset.php

```diff
@@ -58,12 +59,18 @@ public function delete()
     {
         if ($this->exists()) {
 
-            if (!unlink($this->getAbsoluteLocation()))
+            try {
+                Filesystem::remove($this->getAbsoluteLocation());
+            } catch (Exception $e) {
                 throw new Exception("Unable to delete merged file : " . $this->getAbsoluteLocation() . ". Please delete the file and refresh");
+            }
 
             // try to remove compressed version of the merged file.
-            @unlink($this->getAbsoluteLocation() . ".deflate");
-            @unlink($this->getAbsoluteLocation() . ".gz");
+            try {
+                Filesystem::remove($this->getAbsoluteLocation() . ".deflate");
+                Filesystem::remove($this->getAbsoluteLocation() . ".gz");
+            } catch (Exception $e) {
+            }
```

Comment:
No, but I would assume in most cases we wouldn't want a failed remove to result in an exception that causes the request to fail. So there may be custom error handling, and using a try-catch every time for something as simple as file deletion seems verbose to me.